src/framework/theme/components/tree-grid/data-source/tree-grid-data-source.ts

```diff
@@ -130,15 +140,20 @@ export class NbTreeGridDataSourceBuilder<T> {
               private treeGridDataService: NbTreeGridDataService<T>) {
   }
 
-  create(data: NbTreeGridNode<T>[]): NbTreeGridDataSource<T> {
+  create<N>(
+    data: N[],
+    dataGetter?: DataGetter<N, T>,
```

Comment:
Let's configure all these getters through an optional object with optional fields. It would be much more convenient in cases when you want to provide *expandedGetter* only and don't want to provide others.

Reply:
Something like this:
```
  interface Getters {
    dataGetter?: DataGetter<N, T>,
    childrenGetter?: ChildrenGetter<N, T>,
    expandedGetter?: ExpandedGetter<T>,
  }

  create<N>(
    data: N[],
    getters?: Getters
  ): NbTreeGridDataSource<T> {
  }
```

Reply:
Yeah, that's much better. Thanks!